### PR TITLE
[unit_converter]: Update unit_converter to API v2.3

### PR DESCRIPTION
**Pull request by Pete-Hamlin**
Reopening of https://github.com/albertlauncher/python/pull/190

### CODE REVIEW

**Comment by hfr-hfr** [comment]
@ManuelSchneid3r is there anything we can do to help this getting merged? Is there any issue with it?

**Comment by ManuelSchneid3r** [comment]
you could update it to 2.3

**Comment by Pete-Hamlin** [comment]
Have updated extension to API v2.3 and have tested it working locally.

Had to nuke my `site-packages` and do a clean install of dependencies for working with the new API, but seem to have only had to do this once.

File: unit_converter/__init__.py [hunk 1]
```diff
@@ -334,7 +328,7 @@ class Plugin(albert.TriggerQueryHandler):
             "C": "degC",
         },
         # Display names for units
-        # Units may be added here to override the default display names
+        # Units may be added here to o>verride the default display names
```
Comment (ManuelSchneid3r) [inline unit_converter/__init__.py:331]: probably added accidentially

**Review by ManuelSchneid3r — COMMENTED** [review]
otherwise LGTM

File: unit_converter/__init__.py (hunk 1 above)
Comment (Pete-Hamlin) [reply unit_converter/__init__.py]: You're not wrong.

Fixed in 999c115e34e8d48cba2dec5e83596331837ec1eb

**Comment by Pete-Hamlin** [comment]
Thanks for the feedback, should be all resolved as of c2fcbc5a20027ea6bbc62612225262fd23bd8ed3

Have tested  & all still working.

**Comment by hfr-hfr** [comment]
Thanks for the update, @Pete-Hamlin ! 

Unfortunately I get an error when activating the plug-in:

> Unit Converter (python.unit_converter):
> ImportError: cannot import name 'dataclass_transform' from 'typing_extensions' (/usr/lib/python3/dist-packages/typing_extensions.py)

**Comment by Pete-Hamlin** [comment]
@hfr-hfr can you try deleting/moving your `site-packages` folder and restarting? This is usually my go to if I'm getting weird dependency errors in albert.

Something like `mv .local/share/albert/python/site-packages/ mv .local/share/albert/python/site-packages.bak/` and see if it sorts itself out?

**Comment by hfr-hfr** [comment]
After nuking site-packages and re-installing dependencies for unit_converter, I get a slightly different error:

> Unit Converter (python.unit_converter):
> ImportError: cannot import name 'is_typeddict' from 'typing_extensions' (/usr/lib/python3/dist-packages/typing_extensions.py)

**Comment by ManuelSchneid3r** [comment]
@hfr-hfr please post stdout. It contains a lot of information we would have to ask for.

**Comment by ManuelSchneid3r** [comment]
And open a new issue. This is a merged PR

**Comment by ManuelSchneid3r** [comment]
this plugin is heavily overengineered imho and to be honest i am not keen on maintaining it. if anybody is interested, i'd be happy to merge a PR that drops like 300-400 lines of code.

**Comment by hfr-hfr** [comment]
> And open a new issue. This is a merged PR

https://github.com/albertlauncher/albert/issues/1423#issue-2477929886

**Comment by hfr-hfr** [comment]
Fix is mentioned in issue above, @Pete-Hamlin :pray:

**Pull request merged**